server/models/supplement.ts

```diff
@@ -66,9 +66,21 @@ export class Supplement
 
     public toDB() : Record<string, unknown>
     {
-        const { id, ...obj } = snakeCaseKeys(this.toJSON());
+        const { id, ...suppDef } = snakeCaseKeys(this.toJSON());
+
+        // And nested keys need to become json strings
+        for(const key in suppDef)
+        {
+            const value = suppDef[key];
+
+            if(typeof value === 'object' && value !== null)
+            {
+                suppDef[key] = JSON.stringify(value);
+            } // end if
+        } // end for
+
```

Comment:
I'm not sure I'm getting any this exists. We take an object, extract the id property, transform the remaining property names, and then jsonify the value of each property. Is there a 1:1 mapping between a supplement and the table column fields? If so, why? Why not have a column 'supplement_stuff' and just encode the `suppDef` directly?

Reply:
That's not quite what's going on here. We're not jsonifying every value. We're only jsonifying any _objects_ that are values, because the only way to represent those in the DB is with a json column. Everything else is left alone.

There is a 1:1 mapping, and the reason why I did that was because I was going with the standard wisdom that it's better to have you db tables reflect the structure of your object, where possible. As needing a json column is a rarity (only objects or arrays would need to be encoded that way), this seemed like a sane way to handle it.

Now, you're right, that might be simpler in some areas, though it makes the object a tad more awkward to work with, or if I shred the json, more complicated to encode/decode. There's only one supplement that has a nested object, that's `forcePowers`. I don't think it'll be a common thing, so I'm content with this solution for the time being.